From 88b2632635712e1c2e9896397e2fa4f10c140823 Mon Sep 17 00:00:00 2001
From: Sherman Siu <s8siu@uwaterloo.ca>
Date: Mon, 21 Jul 2025 05:05:36 -0400
Subject: [PATCH 3/6] Manually define gettid

- `gettid` is not defined by default until glibc 2.30: see https://stackoverflow.com/questions/30680550/c-gettid-was-not-declared-in-this-scope for details
---
 csrc/cpu/utils.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/csrc/cpu/utils.cpp b/csrc/cpu/utils.cpp
index fd06e01e4d..c4b62f37ed 100644
--- a/csrc/cpu/utils.cpp
+++ b/csrc/cpu/utils.cpp
@@ -12,6 +12,9 @@
 
 #include "cpu/utils.hpp"
 
+#include <sys/syscall.h>
+#define gettid() syscall(SYS_gettid)
+
 #ifdef VLLM_NUMA_DISABLED
 void init_cpu_memory_env(std::vector<int64_t> node_ids) {}
 #else
-- 
2.51.0

